src/pm/hydra/lib/utils/args.c

```diff
@@ -97,9 +97,11 @@ HYD_status HYDU_find_in_path(const char *execname, char **path)
             MPL_free(path_loc);
             path_loc = NULL;
 
+            MPL_free(test_loc);
             status = get_abs_wd(strtok(NULL, ";:"), &test_loc);
             HYDU_ERR_POP(status, "error getting absolute working dir\n");
         }
+        if (test_loc != NULL) MPL_free(test_loc);
```

Comment:
`MPL_free` checks for `NULL`  thus the `if`-check is unnecessary. 

Also would't it leak when it `goto exit` on line 94? How about move the final free(test_loc) to the `fn_exit` block?

Reply:
The while loop may allocate test_loc multiple time.
It needs to be freed before a new allocation.